### PR TITLE
DashboardManager partial implementation

### PR DESCRIPTION
**Pull request by mmzdouc**
Implementation of GET and POST functionality for dashboard template
See the in-line comments @HAugustijn

### CODE REVIEW

File: fermo_gui/analysis/dashboard_manager.py [hunk 1]
```diff
@@ -20,10 +20,416 @@
...
+        Arguments:
+            f_sess: fermo session file
+        """
+        self.extract_stats_analysis(f_sess)
+        self.extract_stats_samples_dyn(f_sess)
+
+    def provide_data_get(self: Self) -> dict:
+        """Return data required by GET method
+
+        Returns: a json-compatible dict
```
Comment (mmzdouc) [inline fermo_gui/analysis/dashboard_manager.py:54]: @HAugustijn this is the current GET information: a dict of "stats_analysis" for the overall analysis information, and "stats_samples_dyn" which is partially dynamic and can be later updated when any filters are applied

File: fermo_gui/analysis/dashboard_manager.py [hunk 2]
```diff
@@ -20,10 +20,416 @@
...
+            "stats_analysis": self.stats_analysis,
+            "stats_samples_dyn": self.stats_samples_dyn,
+        }
+
+    def extract_stats_analysis(self: Self, f_sess: dict):
+        """Extracts static analysis stats from fermo.session file
+
+        Arguments:
+            f_sess: fermo session file
+        """
```
Comment (mmzdouc) [inline fermo_gui/analysis/dashboard_manager.py:66]: @HAugustijn the individual information in "stats_analysis" - can be iterated over

File: fermo_gui/analysis/dashboard_manager.py [hunk 3]
```diff
@@ -20,10 +20,416 @@
...
+                ),
+            }
+        except TypeError:
+            self.stats_analysis = {"error": "error during parsing of session file"}
+
+    def extract_stats_samples_dyn(self: Self, f_sess: dict):
+        """Extracts dynamic stats of samples
+
+        Arguments:
+            f_sess: fermo session file
```
Comment (mmzdouc) [inline fermo_gui/analysis/dashboard_manager.py:86]: @HAugustijn the individual information in "stats_samples_dyn". "retained_features" is simply the total features per sample on "GET" and is then adjusted to the number of features per samples that are inside the filter settings on POST

**Review by mmzdouc — COMMENTED** [review]
LGTM

**Review by HAugustijn — APPROVED** [review]
LGTM

**Pull request merged**